.github/workflows/codeql.yml

```diff
@@ -0,0 +1,60 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master", "dev" ]
+  pull_request:
+    branches: [ "master", "dev" ]
+  schedule:
```

Comment:
Why do we need to run it on schedule? Why not only in the pull requests?

Reply:
Its a recommendation, since new vulnerabilities may be discovered even if the codebase didn't change.
This will allow github to notify us that the code has a recently discovered vulnerability for instance.